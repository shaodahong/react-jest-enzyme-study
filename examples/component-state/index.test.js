import React from 'react';
import ComponentState from './index';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow, render, mount } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

it('ComponentState test', () => {
  const wrapper = shallow(<ComponentState />);
  expect(wrapper).toMatchSnapshot();

  const text = 'hello monkey';
  wrapper.setState({ text });
  expect(wrapper).toMatchSnapshot();

  expect(wrapper.text()).toBe(text);
});
