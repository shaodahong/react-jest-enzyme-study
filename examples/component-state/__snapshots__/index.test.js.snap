// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentState test 1`] = `<div />`;

exports[`ComponentState test 2`] = `
<div>
  hello monkey
</div>
`;
